docs/api/valueid.md

```diff
@@ -29,3 +29,155 @@ interface TranslatedValueID extends ValueID {
 ```
 
 These properties are meant to provide a human-readable representation of the Command Class, property and property key.
+
+## ValueMetadata
+
+Value metadata is used to get additional information about a specific `ValueID`. All metadata shares the following structure (with additional information added) for each value type:
+
+```ts
+interface ValueMetadataBase {
+	type: ValueType;
+	readable: boolean;
+	writeable: boolean;
+	description?: string;
+	label?: string;
+	ccSpecific?: Record<string, any>;
+}
+```
+
+-   `type`: The type of the value. Can be: `any`, `number`, `boolean`, `string`, `number[]`, `string[]`, `boolean[]`. Depending on the `type`, each metadata can have additional properties (see below).
+-   `readable`: Whether the value can be read. Default: `true`
+-   `writable`: Wheather a value can be written. Default: `true`
+-   `description`: A description of the value
+-   `label`: A human-readable label for the property
+-   `ccSpecific`: CC specific information to help identify this value [(see below)](#CC-specific-fields)
+
+### Value types
+
+Here you can find all the type specific metadata fields
+
+#### `any`
+
+```ts
+interface ValueMetadataAny extends ValueMetadataBase {
+	default?: any;
+}
+```
+
+-   `default`: The default value
+
+#### `number`
+
+```ts
+interface ValueMetadataNumeric extends ValueMetadataBase {
+	type: "number";
+	min?: number;
+	max?: number;
+	steps?: number;
+	default?: number;
+	states?: Record<number, string>;
+	unit?: string;
+}
+```
+
+-   `min`: The minumum value that can be assigned to this value
+-   `max`: The maximum value that can be assigned to this value
+-   `steps`: When only certain values between min and max are allowed, this determines the step size
+-   `default`: The default value
+-   `states`: Human-readable names for numeric values
+-   `unit`: An optional unit for numeric values
+
+#### boolean
+
+```ts
+interface ValueMetadataBoolean extends ValueMetadataBase {
+	type: "boolean";
+	default?: number;
+}
+```
+
+-   `default`: The default value
+
+### string
+
+```ts
+interface ValueMetadataString extends ValueMetadataBase {
+	type: "string";
+	minLength?: number;
+	/** The maximum length this string may have (optional) */
+	maxLength?: number;
+	/** The default value */
+	default?: string;
+}
+```
+
+-   `minLength`: The minimum length this string may have
+-   `maxLength`: The maximum length this string may have
+-   `default`: The default value
+
+### CC-specific fields
+
+The structure of the `ccSpecific` fields is shown here for each CC that provides them:
+
+#### Alarm Sensor CC
+
+```ts
+type AlarmSensorValueMetadata = ValueMetadata & {
+	ccSpecific: {
+		sensorType: AlarmSensorType;
+	};
+};
+```
+
+#### Binary Sensor CC
+
+```ts
+type BinarySensorValueMetadata = ValueMetadata & {
+	ccSpecific: {
+		sensorType: BinarySensorType;
+	};
+};
+```
+
+#### Meter CC
+
+```ts
+type MeterMetadata = ValueMetadata & {
+	ccSpecific: {
+		meterType: number;
```

Comment:
Link to meter types json?

Reply:
https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/meters.json